tests/unit/test_client.py

```diff
@@ -8689,6 +8689,32 @@ def test_load_table_from_dataframe_w_nulls(self):
         assert sent_config.schema == schema
         assert sent_config.source_format == job.SourceFormat.PARQUET
 
+    @unittest.skipIf(pandas is None, "Requires `pandas`")
+    @unittest.skipIf(pyarrow is None, "Requires `pyarrow`")
+    def test_load_table_from_dataframe_w_nulls_for_required_cols(self):
+        """Test that a DataFrame with null columns should throw error if
+        corresponding field in bigquery schema is required.
+
+        See: https://github.com/googleapis/python-bigquery/issues/1692
+        """
+        from google.cloud.bigquery.schema import SchemaField
+        from google.cloud.bigquery import job
+
+        client = self._make_client()
+        records = [{"name": None, "age": None}, {"name": None, "age": None}]
+        dataframe = pandas.DataFrame(records, columns=["name", "age"])
+        schema = [
+            SchemaField("name", "STRING"),
+            SchemaField("age", "INTEGER", mode="REQUIRED"),
+        ]
+        job_config = job.LoadJobConfig(schema=schema)
+        with pytest.raises(ValueError) as e:
+            client.load_table_from_dataframe(
+                dataframe, self.TABLE_REF, job_config=job_config, location=self.LOCATION
+            )
+
+            assert str(e.value) == "required field age can not be nulls"
```

Comment:
When the exception is thrown, it returns to the context manager (pytest) and assertion is never reached. See the test below for an example for testing the message value.

Reply:
oh yes.. fixed it.